### PR TITLE
Commander skill description support

### PR DESCRIPTION
**Pull request by dydzio0614**

### CODE REVIEW

File: client/windows/CCreatureWindow.cpp [hunk 1]
```diff
@@ -521,8 +532,16 @@ void CStackWindow::CWindowSection::createCommanderAbilities()
 	}
 }
 
-void CStackWindow::setSelection(si32 newSkill, CClickableObject * newIcon)
+void CStackWindow::setSelection(si32 newSkill, CCommanderSkillIcon * newIcon)
 {
+	auto getSkillDescription = [this](int skillIndex, bool selected) -> std::string
+	{
```
Comment (DjWarmonger) [inline client/windows/CCreatureWindow.cpp]: This code is duplicated, but could be easily extracted - doesn't event depend on this class.

File: lib/CGeneralTextHandler.cpp [hunk 2]
```diff
@@ -490,4 +490,15 @@ CGeneralTextHandler::CGeneralTextHandler()
 		}
 		while (parser.endLine());
 	}
+	if (VLC->modh->modules.COMMANDERS)
```
Comment (alexvins) [inline lib/CGeneralTextHandler.cpp]: ZNPC00 is part of WoG, but COMMANDERS module is independent. You should handle that somehow.

File: client/windows/CCreatureWindow.cpp [hunk 3]
```diff
@@ -521,8 +532,16 @@ void CStackWindow::CWindowSection::createCommanderAbilities()
...
-void CStackWindow::setSelection(si32 newSkill, CClickableObject * newIcon)
+void CStackWindow::setSelection(si32 newSkill, CCommanderSkillIcon * newIcon)
 {
+	auto getSkillDescription = [this](int skillIndex, bool selected) -> std::string
+	{
+		if(selected)
+			return CGI->generaltexth->znpc00[151 + (12 * skillIndex) + ((info->commander->secondarySkills[skillIndex] + 1) * 2)]; //upgrade description
+		else
+			return CGI->generaltexth->znpc00[151 + (12 * skillIndex) + (info->commander->secondarySkills[skillIndex] * 2)];
+	};
```
Comment (alexvins) [inline client/windows/CCreatureWindow.cpp]: You should handle situation when znpc00 not exists.

File: client/windows/CCreatureWindow.cpp (hunk 1 above)
Comment (dydzio0614) [inline client/windows/CCreatureWindow.cpp]: getSkillDescription actually uses "info" field from CStackWindow. There is also getSkillImage that is currently implemented as code duplicate in CStackWindow::CWindowSection::createCommander and void CStackWindow::setSelection so I follow existing design.

File: lib/CGeneralTextHandler.cpp [hunk 4]
```diff
@@ -490,4 +490,21 @@ CGeneralTextHandler::CGeneralTextHandler()
...
+			parser.endLine();//header
+
+			do
+			{
+				znpc00.push_back(parser.readString());
+			} while (parser.endLine());
+		}
+		catch (std::runtime_error)
+		{
+			logGlobal->error("WoG file ZNPC00.TXT containing commander texts was not found");
```
Comment (alexvins) [inline lib/CGeneralTextHandler.cpp]: This should not be an error.

File: client/windows/CCreatureWindow.cpp [hunk 5]
```diff
@@ -521,8 +538,22 @@ void CStackWindow::CWindowSection::createCommanderAbilities()
...
 
-void CStackWindow::setSelection(si32 newSkill, CClickableObject * newIcon)
+void CStackWindow::setSelection(si32 newSkill, CCommanderSkillIcon * newIcon)
 {
+	auto getSkillDescription = [this](int skillIndex, bool selected) -> std::string
+	{
+		if (CGI->generaltexth->znpc00.size() == 0)
+		{
+			logGlobal->warn("WoG file with commander skill descriptions not found!");
+			return "";
```
Comment (alexvins) [inline client/windows/CCreatureWindow.cpp]: No need to report this

**Comment by dydzio0614** [comment]
Anything still needed before this branch can be merged?

**Comment by ArseniyShestakov** [comment]
Currently attempt to go through broken WoG support and get it merged. :-)

**Pull request merged**